x-pack/plugins/infra/public/pages/link_to/use_node_details_redirect.ts

```diff
@@ -29,37 +47,31 @@ export const useNodeDetailsRedirect = () => {
 
   const appId = useObservable(currentAppId$);
   const getNodeDetailUrl = useCallback(
-    ({
-      nodeType,
-      nodeId,
+    <T extends InventoryItemType>({
+      assetType,
+      assetId,
       search,
-    }: {
-      nodeType: InventoryItemType;
-      nodeId: string;
-      search: QueryParams;
-    }): LinkDescriptor => {
-      const { to, from, ...rest } = search;
+    }: NodeDetailsRedirectParams<T>): LinkDescriptor => {
+      const { from, to, ...additionalParams } = search;
 
       return {
         app: 'metrics',
-        pathname: `link-to/${nodeType}-detail/${nodeId}`,
+        pathname: `link-to/${assetType}-detail/${assetId}`,
         search: {
-          ...rest,
-          ...(to && from
-            ? {
-                to: `${to}`,
-                from: `${from}`,
-              }
+          ...(Object.keys(additionalParams).length > 0
+            ? { [REDIRECT_ASSET_DETAILS_KEY]: rison.encodeUnknown(additionalParams) }
             : undefined),
-          // While we don't have a shared state between all page in infra, this makes it possible to restore a page state when returning to the previous route
-          ...(location.search || location.pathname
-            ? {
-                state: JSON.stringify({
-                  originAppId: appId,
-                  originSearch: location.search,
-                  originPathname: location.pathname,
-                } as RouteState),
-              }
+          // retrocompatibility
+          ...(from ? { [REDIRECT_NODE_DETAILS_FROM_KEY]: `${from}` } : undefined),
+          ...(to ? { [REDIRECT_NODE_DETAILS_TO_KEY]: `${to}` } : undefined),
+        },
+        state: {
+          ...(location.pathname
+            ? ({
+                originAppId: appId,
+                originSearch: location.search,
+                originPathname: location.pathname,
+              } as RouteState)
```

Comment:
This won't go in the querystring anymore, creating a shorter and less polluted query param.